TeachEasy/src/teacheasy/data/AnswerBox.java

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright (C) 2015 Sofia Software Solutions
+ */
+package teacheasy.data;
+
+import javafx.event.ActionEvent;
+import javafx.event.EventHandler;
+import javafx.scene.Group;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.TextField;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.KeyEvent;
+import javafx.scene.layout.HBox;
+
+/**
+ * Class to draw answer boxes with the specified x & y location and character
+ * limit. The class also checks entered answer against list of possible answers
+ * and assigns them the predefined marks.
+ * 
+ * @author Daniel Berhe and Jake Ransom
+ * @version 1.0
+ * 
+ * 
+ */
+public class AnswerBox {
+
+	private int marks;
+	private int awardedMarks;
+	private boolean retry;
+	private String correctAnswers;
+	private boolean answerIsCorrect;
+	private int characterLimit;
+	private double xStart;
+	private double yStart;
+	private boolean isNumerical;
+	private boolean validInput;
+
+	private Group group;
+	private TextField answerField;
+	private HBox box;
+	private Button checkAnswerButton;
+	private Label feedbackLabel;
+
+	public AnswerBox(double nXStart, double nYStart, int nCharacterLimit,
+			boolean nRetry, String nCorrectAnswers, int nMarks,
+			boolean nIsNumerical, Group nGroup) {
+
+		answerField = new TextField();
+		/* Setting local variables */
+		this.marks = nMarks;
+		this.correctAnswers = nCorrectAnswers;
+		this.characterLimit = nCharacterLimit;
+		this.xStart = nXStart;
+		this.yStart = nYStart;
+		this.group = nGroup;
+		this.retry = nRetry;
+		this.isNumerical = nIsNumerical;
+
+		box = new HBox();
+
+		/* Creates new button to check the typed answer */
+		checkAnswerButton = new Button("Check answer");
+		checkAnswerButton.setId("check answer");
+		checkAnswerButton.setOnAction(new ButtonEventHandler());
+
+		/* Creates feedback label to display correct/incorrect */
+		feedbackLabel = new Label();
+
+		createAnswerBox(characterLimit, nRetry);
+
+		/* Submits and checks typed answer upon key press "enter" */
+		answerField.setOnKeyPressed(new EventHandler<KeyEvent>() {
```

Comment:
Key event handler could be a class at the end of the file like the button event handler. This would be more readable than the anonymous inner class method used here.